chore: Adds Getting started tutorial (#3)
canonical/chaos-engineering: docs/tutorial/getting_started.md

# Getting started

> In this tutorial, we will deploy and run Charmed Chaos Engineering platform using Juju and Terraform.

Reviewer: ```suggestion
In this tutorial, we will deploy and run the Charmed Chaos Engineering platform using Juju and Terraform.
```

To complete this tutorial, you will need a machine which meets the following requirements:

- 8GB of RAM
- 50GB of free disk space
- An Ubuntu (or another operating system which supports [snapd]) environment to run the commands

## 1. Install Canonical K8s

From your terminal, install Canonical K8s and bootstrap it:

```shell

> sudo snap install k8s --classic --channel=1.33-classic/stable

Reviewer: why not strict? (fine if we add a note to explain why)

cat << EOF | sudo k8s bootstrap --file -
containerd-base-dir: /opt/containerd
disable-rbac: true
cluster-config:
  network:
    enabled: true
  dns:
    enabled: true
  local-storage:
    enabled: true
EOF
```

## 2. Bootstrap a Juju controller

> From your terminal, install Juju.

Reviewer: instead of repeating 'from your terminal', we should add a note at the beginning explaining "all the code blocks in this tutorial can be executed from a linux shell"

```shell
sudo snap install juju --channel=3.6/stable
```

> Save the K8s credentials to allow bootstrapping Juju controller.

Reviewer: ```suggestion
Save the K8s credentials, we will need them later to bootstrap the Juju controller.
```

```shell
mkdir -p ~/.kube
sudo k8s config > ~/.kube/config
mkdir -p ~/.local/share/juju/
sudo k8s config > ~/.local/share/juju/credentials.yaml
```

Bootstrap a Juju controller

```shell
juju bootstrap k8s
```

## 3. Install Terraform

> From your terminal, install Terraform.

Reviewer: ```suggestion
Install Terraform with:
```

```shell
sudo snap install terraform --classic
```

> ## 4. Deploy Charmed Chaos Engineering platform

Reviewer: ```suggestion
## 4. Deploy the Charmed Chaos Engineering platform
```

On the host machine create a new directory called `terraform`:

```shell
mkdir terraform
```

> Inside newly created `terraform` directory create a `versions.tf` file:

Reviewer: ```suggestion
Inside the newly created `terraform` directory create a `versions.tf` file:
```

```shell
cd terraform
cat << EOF > versions.tf
terraform {
  required_providers {
    juju = {
      source  = "juju/juju"
      version = ">= 0.14.0"
    }
  }
}
EOF
```

> Create a Terraform module for Charmed Chaos Engineering platform:

Reviewer: ```suggestion
Create a new Terraform module:
```

```shell
cat << EOF > main.tf
resource "juju_model" "charmed-chaos" {
  name = "charmed-chaos"
}

module "charmed-litmus" {
  source = "git::https://github.com/canonical/litmus-operators//terraform"

  model      = juju_model.charmed-chaos.name
  depends_on = [juju_model.charmed-chaos]
}
EOF
```

```{note}
You can get a ready example by cloning [this Git repository](https://github.com/canonical/chaos-engineering).
All necessary files are in the `examples/terraform/getting_started` directory.
```

> Initialize Juju Terraform provider:

Reviewer: ```suggestion
Initialize the Juju Terraform provider:
```

```shell
terraform init
```

> Deploy Charmed Chaos Engineering platform by applying your Terraform configuration:

Reviewer: ```suggestion
Apply your Terraform configuration:
```

```shell
terraform apply -auto-approve
```

<!--
This hidden block is here for documentation testing purposes and will not be rendered in the official documentation.

> ```shell

Reviewer: cool!

mongodb-k8s/2          active    idle   10.1.194.230         

Integration provider           Requirer                             Interface                Type     Message
litmus-auth:http-api           litmus-chaoscenter:auth-http-api     litmus_auth_http_api     regular  
litmus-auth:litmus-auth        litmus-backend:litmus-auth           litmus_auth              regular  
litmus-backend:http-api        litmus-chaoscenter:backend-http-api  litmus_backend_http_api  regular  
mongodb-k8s:database           litmus-auth:database                 mongodb_client           regular  
mongodb-k8s:database           litmus-backend:database              mongodb_client           regular  
mongodb-k8s:database-peers     mongodb-k8s:database-peers           mongodb-peers            peer     
mongodb-k8s:ldap-peers         mongodb-k8s:ldap-peers               ldap-peers               peer     
mongodb-k8s:status-peers       mongodb-k8s:status-peers             status-peers             peer     
mongodb-k8s:upgrade-version-a  mongodb-k8s:upgrade-version-a        upgrade                  peer
```

## 5. Log in to the Litmus ChaosCenter

> Retrieve the ChaosCenter IP address from `juju status` output. In this tutorial this is `10.1.194.255`.

Reviewer: ```suggestion
Retrieve the `chaoscenter/0` unit IP address from the `juju status` output. In the example above, this is `10.1.194.255`; yours will likely differ.
```

> In your browser navigate to `http://{CHAOSCENTER_IP}:8185`. You should see Litmus login page:

Reviewer: ```suggestion
In your browser navigate to `http://{CHAOSCENTER_IP}:8185`. You should see the Litmus login page:
```
